### ISSUE TITLE
how to save and load the model

### ISSUE THREAD

**Issue by liguheng** [huggingface/candle#2916]
I  just use the varmap.save the varmap,but when I use the varmap.load then achieved a empty varmap. is there any way to save the trained model?

**Comment by greenrazer** [comment]
Could you share a minimal example of how you're saving and loading the varmap?

**Comment by liguheng** [comment]
> Could you share a minimal example of how you're saving and loading the varmap?
that's the example:
///
fn main() {
    let dev=candle_core::Device::cuda_if_available(0).unwrap();
    let varmap=candle_nn::VarMap::new();
    let vb=candle_nn::VarBuilder::from_varmap(&varmap, candle_core::DType::F32, &dev);
    let dataset=DataSet::read("/home/liguheng/sharefile/temp0.csv","/home/liguheng/sharefile/temp.csv",&dev);
    let model=Mlstm::new(50, 32, 3, (dataset.n/1000*1000+2000) as usize, vb);
    let mut opt=candle_nn::optim::AdamW::new_lr(varmap.all_vars(), 1e-1).unwrap();
    let data=dataset.data;
    let label=dataset.label;
    for _ in 0..1 {
        let logits=model.forward(&data);
        let loss=candle_nn::ops::log_softmax(&logits, candle_core::D::Minus1).unwrap();
        let loss=candle_nn::loss::nll(&loss, &label).unwrap();
        opt.backward_step(&loss).unwrap();
        println!("{}",loss.to_scalar::<f32>().unwrap());
    }
    let result=model.forward(&dataset.test);
    let result=result.argmax(candle_core::D::Minus1).unwrap();
    println!("result:{}",result);
    println!("varmap_to_save:{:?}",varmap.all_vars());
    varmap.save("model").unwrap();
    let mut load_varmap=candle_nn::VarMap::new();
    load_varmap.load("model").unwrap();
    println!("varmap_loaded:{:?}",load_varmap.all_vars());
}
///
that's the result it print:
///
1.1250305
result:[0]Tensor[[1], u32]
varmap_to_save:[Var(Tensor[dims 16; f32]), Var(Tensor[dims 128, 50; f32]), Var(Tensor[dims 128; f32]), Var(Tensor[dims 128, 32; f32]), Var(Tensor[0.20531547, 0.25744477, -0.09718387; f32]), Var(Tensor[dims 3, 16; f32]), Var(Tensor[dims 13000, 50; f32]), Var(Tensor[dims 16, 32; f32]), Var(Tensor[dims 128; f32])]
varmap_loaded:[]
//

**Comment by liguheng** [comment]
i have seen this https://github.com/huggingface/candle/issues/1469#issue-2054672858,i have figure it out through pytorch docs,thanks
